python/transcriber/transcriber.py

```diff
@@ -0,0 +1,131 @@
+            for i, res in enumerate(result):
+                if not 'result' in res:
+                    continue
+                words = res['result']
+                for j in range(0, len(words), WORDS_PER_LINE):
+                    line = words[j : j + WORDS_PER_LINE]
+                    s = srt.Subtitle(index=len(subs),
+                            content = ' '.join([l['word'] for l in line]),
+                            start=datetime.timedelta(seconds=line[0]['start']),
+                            end=datetime.timedelta(seconds=line[-1]['end']))
+                    subs.append(s)
+        final_result = ''
+        if args.outputtype == 'srt':
+            final_result = srt.compose(subs)
+        elif args.outputtype == 'txt':
+            for i in range(len(result)):
+                final_result += result[i]['text'] + ' '
+        return final_result, tot_samples
+
+    def resample_ffmpeg(infile):
+        process = subprocess.Popen(
+            ['ffmpeg', '-nostdin', '-loglevel', 'quiet', '-i', 
+            infile, 
+            '-ar', '16000','-ac', '1', '-f', 's16le', '-'], 
+            stdout=subprocess.PIPE)
+        return process
+
+    def get_time():
+        start_time = dt.now()
+        return start_time
+
+    def send_time(start_time):
+        script_time = str(dt.now() - start_time)
+        seconds = script_time[5:8].strip('0')
+        mcseconds = script_time[8:].strip('0')
+        return script_time.strip(':0'), seconds.rstrip('.'), mcseconds
+
+    def process_dir(args):
+        files = os.listdir(args.input)
+        arg_list = list()
+        input_dir = args.input + '/'
+        output_dir = args.output + '/'
+        extension_i = f".{files[0].split('.')[1]}"
+        format_o = f".{args.outputtype}"
+        input_list = [input_dir + each for each in files]
+        output_list = [output_dir + each.replace(extension_i, format_o) for each in files]
+        [arg_list.extend([(input_list[i], output_list[i])]) for i in range(len(input_list))]
+        return arg_list
+
+    def get_list_models():
+        response = requests.get(model_list_url)
+        [print(response.json()[i]['name']) for i in range(len(response.json()))]
+        exit(1)
+
+    def get_model(args):
+
+        def get_model_path():
+            model_path = Path.home() / '.cache' / 'vosk'
+            if not Path.is_dir(model_path):
+                Path.mkdir(model_path)
+            return model_path
+
+        def download_model(result_model, model_path):
+            model_zip = Path(model_path, result_model + '.zip')
+            model_url = model_pre_path + result_model + '.zip'
+            urllib.request.urlretrieve(model_url, model_zip)
+            with zipfile.ZipFile(model_path / model_zip, 'r') as model_ref:
+                model_ref.extractall(model_path)
+            Path.unlink(model_path / model_zip)
+
+        model_list = list()
+        response = requests.get(model_list_url)
```

Comment:
No need to query model list every time you run the tool, only when you want to download the model